### ISSUE TITLE
Focus loss issues with xwayland app + doesn't show up in window switcher

### ISSUE THREAD

**Issue by Flrian** [labwc/labwc#1167]
Sending (+follow) JDownloader to another desktop, or closing a child window makes it lose (or not regain?) focus. Also its not showing up in the window switcher osd. Started after  #1163.

https://0x0.st/H46r.mkv

**Comment by jlindgren90** [comment]
I'm not sure why this would have shown up only after #1163 but it's probably at least somewhat related to #1103.

xprop says:
```
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: False
		Initial state is Normal State.
WM_PROTOCOLS(ATOM): protocols  WM_TAKE_FOCUS, WM_DELETE_WINDOW
```

However the JDownloader window does not paint correctly for me so I'm unable to debug this further.

**Comment by jlindgren90** [comment]
Just found this on the Arch Wiki (https://wiki.archlinux.org/title/wayland):

> The _JAVA_AWT_WM_NONREPARENTING [environment variable](https://wiki.archlinux.org/title/Environment_variable) can be set to "1" to fix misbehavior where the application starts with a blank screen.

**Comment by jlindgren90** [comment]
Related: #1130 and #1137.

**Comment by johanmalm** [comment]
> Just found this on the Arch Wiki (https://wiki.archlinux.org/title/wayland):
> 
> > The _JAVA_AWT_WM_NONREPARENTING [environment variable](https://wiki.archlinux.org/title/Environment_variable) can be set to "1" to fix misbehavior where the application starts with a blank screen.

See this too:

https://github.com/labwc/labwc/blob/50ceae5fb4eb033dbffc04cf41eabc920727dd11/docs/environment#L25-L28

**Comment by jlindgren90** [comment]
> closing a child window makes it lose (or not regain?) focus

I have an idea how to fix this part, will submit a PR later on.
The window not showing up in Alt-Tab would be addressed by #1139.
Not sure about the loss of focus when sending to another desktop.
